source/braille.py

```diff
@@ -1729,15 +1738,23 @@ def _doCursorMove(self, region):
 		region.update()
 		self.mainBuffer.update()
 		self.mainBuffer.restoreWindow()
-		if region.brailleCursorPos is not None:
-			self.mainBuffer.scrollTo(region, region.brailleCursorPos)
-		elif region.brailleSelectionStart is not None:
-			self.mainBuffer.scrollTo(region, region.brailleSelectionStart)
+		self.scrollToCursorOrSelection(region)
 		if self.buffer is self.mainBuffer:
 			self.update()
 		elif self.buffer is self.messageBuffer and keyboardHandler.keyCounter>self._keyCountForLastMessage:
 			self._dismissMessage()
 
+	def scrollToCursorOrSelection(self, region):
+		if region.brailleCursorPos is not None:
+			self.mainBuffer.scrollTo(region, region.brailleCursorPos)
+		elif not isinstance(region, (NVDAObjectRegion, TextInfoRegion)) or not region.obj.isTextSelectionAnchoredAtStart:

```

Comment:
Motivation for this test:
- If a region is an NVDAObjectRegion it will have an NVDAObject, which has the isTextSelectionAnchoredAtStart attribute.
- If a region is a TextInfoRegion then it has a TextInfo and it will be based on a TreeInterceptor. DocumentTreeInterceptor is the only TreeInterceptor with a TextInfo, so the region must have one of those as its object. This also means the selection anchor attribute is present.